### DISCUSSION TITLE
关于生成图片的几个提问方式

### DISCUSSION THREAD

**Discussion by KonghaYao** [thinkingjimmy/Learning-Prompt#50] [Ideas]
我最近也在研究如何使用 OpenAI 的接口推进 我的 AI 绘画工作，我在我的项目里面使用了下面的一些提问方式，并且直接使用字符串模板的方式载入到了我的 AI 绘图网站中。经过我几天的研究，发现 AI 描述物体的能力强于 Tags 组合，但是AI一般不会出头部的风格组合词。希望能够帮到你。虽然是中文提问，也是可以回复很高准确度的。然后 我用的是 StableDiffusion，并不是 MJ，但我猜测是可以通用的。

![image](https://user-images.githubusercontent.com/68177907/227897414-8986d452-d453-4cb7-971d-10f087371e37.png)

https://gitee.com/dongzhongzhidong/magic-tag/blob/master/Chat_Prompt.md

**Comment by thinkingjimmy** [comment]
感谢分享，我最近也正在肝 MJ 相关的教程 🥰

**Comment by KonghaYao** [comment]
其实 AI 绘画这方面吧，应该按照模型来划分的，MJ 有 Niji 和 官版嘛；SD 就更多了，C 站上面有很多模型，模型适配的风格又不太一样，导致其实有些东西是画不太好的😂。

一般 SD 有根据官方的美式风格的模型、根据 NovelAI 改进的版本（二次元基本上都是）和一个 ChillOut（貌似是另外训练的），这三大方向基本可以搞定很多风格了。

然后对于 SD prompt 的编写方面，我的网站 [魔导绪论](https://magic-tag.netlify.app/#webview) 的长期使用经验是 使用 Tag 定制风格，然后第二段使用 AI 生成的文本进行主体的描述，如果需要细节，使用第三段 Tag 进行补充。

使用 Tag 是因为可以多重复用，而 ChatGPT 生成的风格词其实会携带很多感觉方面的描述，对于主体的描述就一句中文直接生成即可。
如果你感兴趣可以看看我的画廊里面的，都是 SD 这方面的成品😄。

https://magic-tag.netlify.app/gallery/?r18=true

![image](https://user-images.githubusercontent.com/68177907/230327715-11b8c951-2ba1-4861-b465-d28406a27916.png)

**Reply by KonghaYao** [reply]
@thinkingjimmy 画廊里面可以看到每一张图片的提示词，或许可以把 画廊 里面的 提示词给 MJ 试试，或许有奇效😂。

**Reply by thinkingjimmy** [reply]
嗯 我最近也打算研究下 SD ，不过就以功能、迭代和社区支持来看，我觉得 Midjourney 比 SD 要好（纯主观看法）

**Reply by KonghaYao** [reply]
我个人认为这两个各有优劣，SD 开源模型多，辅助插件多，自由度高；MJ 质量高，调教好，在线部署轻松。对于 PE 来说，其实只要关注 如何编写 prompt 就可以了✒️。

AI 绘画太广了，额外的修复手段就有很多种，但主要的画面影响因素是 prompt，其余的手段都算是锦上添花。据我研究，好的 prompt 的稳定性和质量是非常高的，步数和参数都比较差的情况下，复现率也可以达到 70% 以上，同时其部分信息的可迁移性很高，可以快速抽离和使用。所以我会进行 prompt 的脱水，产出基础的元信息复用，下次写起来就很快🥳。

**Reply by thinkingjimmy** [reply]
对，各有优劣。我最近用了下 SD，我发现我更适合 MJ 😂
